packages/lesswrong/server/users/permanentDeletion.ts

```diff
@@ -126,4 +127,32 @@ async function permanentlyDeleteUserById(userId: string) {
   await permanentlyDeleteUser(user)
 }
 
+const cutoffOffsetMs = ACCOUNT_DELETION_COOLING_OFF_DAYS * 24 * 60 * 60 * 1000;
+
+addCronJob({
+  name: "permanentlyDeleteUsers",
+  interval: "every 1 hour",
+  job: async () => {
+    const deletionRequestCutoff = new Date(Date.now() - cutoffOffsetMs)
+
+    const usersToDelete = await Users.find({ permanentDeletionRequestedAt: { $lt: deletionRequestCutoff } }).fetch();
+
+    if (usersToDelete.length > 10) {
```

Comment:
Maybe the limit should be a DB setting? I'm just a bit worried that if we every actually hit that limit then the whole feature would be broken without changing this code (plus I'm not sure how quickly we'd notice).

Reply:
Given that this runs every hour I think this limit is quite high (and we would only hit it if the same users are repeatedly failing), and if we set up an alert we should be able to catch it before/shortly after it hits it